File: 2744-find-maximum-number-of-string-pairs/2744-find-maximum-number-of-string-pairs.cpp
class Solution {
public:
    int maximumNumberOfStringPairs(vector<string>& words){
        int n=words.size();
        set<string> s;
        int ans=0;
        for(auto x:words){//each word
            string st = x;
            reverse(begin(x),end(x));
            if(!s.count(x)){
                s.insert(st);
            }
            else ans++;
        }
        return ans;
    }
};

File: 2744-find-maximum-number-of-string-pairs/2744-find-maximum-number-of-string-pairs.js
/**
 * @param {string[]} words
 * @return {number}
 */
var maximumNumberOfStringPairs = function(words) {
    let n=words.length;
    s = new Set();
    let ans=0;
    for(const x of words){
       let y = x;
       let res = x.split("").reverse().join("");
       if(!s.has(res)){
           s.add(y);
       }
        else ans++;
    }
    return ans;
};